botocore/utils.py

```diff
@@ -1569,6 +1571,45 @@ def set_endpoint(self, request, **kwargs):
     def _use_accesspoint_endpoint(self, request):
         return 's3_accesspoint' in request.context
 
+    def _validate_fips_supported(self, request):
+        client_region = request.context.get('client_region', '')
+        if 'fips' not in client_region:
+            return
+        if 'outpost_name' in request.context['s3_accesspoint']:
+            raise UnsupportedS3AccesspointConfigurationError(
+                msg=(
+                    'Invalid configuration Outpost Access Points '
+                    'do not support Fips- regions'
+                )
+            )
+        client_region = client_region.replace('fips-', '')
+        if request.context['s3_accesspoint']['region'] != client_region:
+            if self._s3_config.get('use_arn_region'):
+                raise UnsupportedS3AccesspointConfigurationError(
+                    msg=(
+                        'Invalid configuration, '
+                        'FIPS region does not match ARN region'
+                    )
+                )
+            else:
+                raise UnsupportedS3AccesspointConfigurationError(
+                    msg=(
+                        'Invalid configuration, cross region Access Point ARN'
+                    )
+                )
+
+    def _validate_global_regions(self, request):
+        if self._s3_config.get('use_arn_region'):
+            return
+        client_region = request.context.get('client_region', '')
+        if client_region in ['aws-global', 's3-external-1']:
+            raise UnsupportedS3AccesspointConfigurationError(
+                msg=(
+                    'Invalid configuration, client region '
+                    'is not a regional endpoint'
+                )
```

Comment:
For all of the error messages that involve using incompatible region, it may be good to say in the error message what the client region that was received so it is easier to troubleshoot.